### PR TITLE
Sound enhancement

### PR DESCRIPTION
**Pull request by RedRafe**
## Brief description of the changes:

Added sound effects as enhanced feedbacks on some player actions:

- on special captain game mode enabled
- during 10s of captain's starting countdown
- on map rerolled
- on player join a team/spectate
- on poll ended
- on @ and whispers pings
- on commands: /where, /inventory, /load-quickbar

Players can choose to have the additional sound FX disabled by switching ON/OFF their own settings. Default will be **ON** (sounds FX are generally very short and should not overlap with other in-game sounds, faking in-game major events such as biter attacks, game lost/won, research completed etc etc....)

Screenshot from 2024-07-12 10-47-31

> Note: when rerolling a new map & charting the whole width, the API caps on chunk requested so during the initial forced charting time, sounds are played at the current chart location and are not audible from players unless the charted chunk is same of player's and/or players are in map view at current chunk charted. This is a limitation of the current sound API but could be improved in 2.0 with new system.

## Tested Changes:
- [x] I've tested the changes locally in SP
- [ ] I've not tested the changes in MP

## Future development
I realize that sound FX may be very subjective. This is a proposed enhancements, but sounds can be added/removed based on game feedbacks from players. 
Furthermore, with the perspective of making the experience better for everyone and not clutter the chat/audio/UI, if these changes are welcomed, we could override the console chat to use different sounds instead. For example, when a player is assigned to a force, instead of printing the message in chat "player has been assigned to X" and playing the join sound, we could merge the two and do `game.print(message, {sound = join_sound})`, which will directly play the join sound on the message instead of default chat message sound. In this way, we both reduce the chat clutter (which I find quite overwhelming when playing as every message/notification/ping/event has the exact same chat sound, so I end up ignoring most of them when playing) and enhance sound FX for a better experience overall.

### CODE REVIEW

File: comfy_panel/config.lua [hunk 1]
```diff
@@ -348,6 +357,10 @@ local function add_switch(element, switch_state, name, description_main, descrip
     label.style.font = 'heading-3'
     label.style.font_color = {0.85, 0.85, 0.85}
 
+    if tooltip then
```
Comment (clifffrey) [inline comfy_panel/config.lua:360]: I would just put this in the `t.add({ type = 'label', caption = description, tooltip = tooltip})` line above, no need for the `if` that way

**Review by clifffrey — CHANGES_REQUESTED** [review]
As a random-thought, if we wanted to fix the charting during map load, instead of unconditionally charting the whole area right at the beginning, we could only chart some of the map (like one column or something), and then schedule more charting to be done inside of either `on_chunk_charted` or `on_chunk_generated` or something. (and we could, in that scheduling, also schedule charting of spectator island again each time)

But, not clear to me that it is actually worth fixing that issue... just wanted to throw out the idea.

I almost always play muted, so I don't personally have much of an opinion about this change, but it seems reasonable to me and the code looks good.

**Comment by RedRafe** [comment]
@clifffrey IIRC some players reported that charting the full 4k width was quite demanding on their PC during the 30s reroll, so maybe this will be a good follow-up PR on its own.
I personally like having the map fully revealed on both sides at the start, aesthetically and functionally, as it benefits both spectators and players to have the map view of it all.

I did some testing when trying to fix the sound bug thing, and my approach was queuing chunk requests every X time during a longer period (using the Task system) instead of a bulk request as it is now. The issue was that if the map gets voted to reroll again, then the requests from the previous map & current map would overlap & get messy. Maybe I should implement a separate queue that also gets flushed upon map reroll so they wouldn't overlap. But yeah, a lot of changes... I'd like to have some inputs from the community as well, just to understand where we want to go. Then implementing will be straightforward

File: comfy_panel/config.lua (hunk 1 above)
Comment (RedRafe) [reply comfy_panel/config.lua]: Good catch, I didnt even notice it was already there. Ty!

**Review by clifffrey — APPROVED** [review]

**Pull request merged**